src/server/game/Server/Packets/ChatPackets.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef ChatPackets_h__
+#define ChatPackets_h__
+
+#include "Packet.h"
+#include "ObjectGuid.h"
+
+namespace WorldPackets
+{
+    namespace Chat
+    {
+        class Emote final : public ServerPacket
+        {
+        public:
+            Emote() : ServerPacket(SMSG_EMOTE, 4 + 8) { }
+
+            WorldPacket const* Write() override;
+
+            int32 EmoteID = 0;
```

Comment:
should this be uint32 ?

Reply:
yes. Even the client magic enum says it's uint32:

https://gist.github.com/Ovahlord/47695c854d14a26215de77e4c3a29e40#file-gistfile1-hpp-L5815

Reply:
no, its signed in client (but it doesnt matter because as you can see in the handler, only a few values are acceptable)